src/libxtp/gwbse/rpa.cc

```diff
@@ -91,6 +91,74 @@ Eigen::MatrixXd RPA::calculate_epsilon(double frequency) const {
 template Eigen::MatrixXd RPA::calculate_epsilon<true>(double frequency) const;
 template Eigen::MatrixXd RPA::calculate_epsilon<false>(double frequency) const;
 
+Eigen::MatrixXd RPA::calculate_epsilon_r(std::complex<double> frequency) const {
+
+  const Index size = _Mmn.auxsize();
+  Eigen::MatrixXd result = Eigen::MatrixXd::Identity(size, size);
+
+  const Index lumo = _homo + 1;
+  const Index n_occ = lumo - _rpamin;
+  const Index n_unocc = _rpamax - lumo + 1;
+
+#pragma omp parallel for schedule(dynamic) reduction(+ : result)
+  for (Index m_level = 0; m_level < n_occ; m_level++) {
+    const double qp_energy_m = _energies(m_level);
+    const Eigen::MatrixXd Mmn_RPA = _Mmn[m_level].bottomRows(n_unocc);
+    const Eigen::ArrayXd deltaE = _energies.tail(n_unocc).array() - qp_energy_m;
+
+    Eigen::VectorXd chi;
```

Comment:
only define chi when you use it, i.e. further below

Reply:
The whole `calculate_epsilon_complex` function is never used. Will remove.